cmd/keycmd/create.go

```diff
@@ -0,0 +1,76 @@
+package keycmd
+
+import (
+	"errors"
+
+	"github.com/ava-labs/avalanche-cli/pkg/key"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/spf13/cobra"
+)
+
+const (
+	forceFlag = "force"
+)
+
+var (
+	forceCreate bool
+	filename    string
+)
+
+func createKey(cmd *cobra.Command, args []string) error {
+	keyName := args[0]
+
+	if app.KeyExists(keyName) && !forceCreate {
+		return errors.New("key already exists. Use --" + forceFlag + " parameter to overwrite")
+	}
+
+	if filename == "" {
+		// Create key from scratch
+		ux.Logger.PrintToUser("Generating new key...")
+		k, err := key.NewSoft(0)
+		if err != nil {
+			return err
+		}
+		keyPath := app.GetKeyPath(keyName)
+		if err := k.Save(keyPath); err != nil {
+			return err
+		}
+		ux.Logger.PrintToUser("Key created")
+	} else {
+		// Load key from file
+		// TODO add validation that key is legal
+		ux.Logger.PrintToUser("Loading user key...")
+		if err := app.CopyKeyFile(filename, keyName); err != nil {
+			return err
+		}
+		ux.Logger.PrintToUser("Key loaded")
+	}
+
+	return nil
+}
+
+func newCreateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "create [keyName]",
+		Short:        "Create a signing key",
+		Long:         `Generates a new private key to use for creating and controlling test subnets.`,
+		Args:         cobra.ExactArgs(1),
+		RunE:         createKey,
+		SilenceUsage: true,
+	}
+
+	cmd.Flags().StringVar(
+		&filename,
+		"filename",
+		"",
```

Comment:
I wonder if we should actually suggest a default path name?

Reply:
That's not a suggestion, that's the value that `filename` gets if there's no flag provided I believe.

Reply:
I understood this is the filename where the key will be saved, not from where it would be imported. Maybe this should be clarified in the docs.